src/hooks/auth.ts

```diff
@@ -0,0 +1,23 @@
+import { BeforeRequestHook } from "./types";
+
+export const injectAPIKey: BeforeRequestHook = {
+    beforeRequest: async (_, request) => {
+        const authz = request.headers.get("Authorization");
+        if (authz) {
+            return request;
+        }
+
+        let token = "";
+        if (typeof process !== "undefined") {
+            token = process.env["DUB_API_KEY"] ?? "";
+        }
+
+        if (!token) {
+            throw new Error("API key is required");
+        }
```

Comment:
I'm not sure if this appropriate behaviour. With this block of code, I'm saying:

> If the original request doesn't have an Authorization header set AND the environment variable `DUB_API_KEY` is not set then fail loudly because the API cannot be accessed without a token.

It might be that you want to let the request go through and have the server respond with 401/403 in which case you might decide to change this block to:

```ts
if (!token) {
  return request
}
```

Reply:
We're okay with failing loudly when the token is missing. 

I'll slightly modify the error message to make it more clear that the token is missing. 

Something like ` "The DUB_API_KEY environment variable is missing or empty; either provide it, or instantiate the Dub client with an token option, like new Dub({ token: 'My Token' })."`